_git/git_commands.py

```diff
@@ -0,0 +1,434 @@
+from aenea import Text
+
+'''
+NOTE: Not all possible commands and options are available in this grammar.
+NatLink puts a limit on how complex the grammar can get, and some commands had
+to be removed. If you feel like there is something missing here, feel free to
+add stuff back and make a pull request.
+
+NOTE 2: Lists of options were grepped from the Git help pages using a bash
+script. These have been marked with the comment 'Generated'. Some of these
+generated lists are too long, so had to be shortened (see note above). You can
+find the script here in case you find it useful
+https://gist.github.com/dylan-chong/ecf701b1a623c9f2ccb78cbb5db700c6 . Ideally,
+when adding a new command, just add the options that are commonly used.
+'''
+
+
+# Common refs for convenience. The user will still have to type out most branch
+# and remote names themselves
+_COMMON_BRANCH_NAMES = ['master', 'develop', 'HEAD']
+_COMMON_REMOTE_NAMES = ['origin', 'upstream']
+
+
+def _add_common_refs(rule_builder):
+    for remote in _COMMON_REMOTE_NAMES:
+        for branch in _COMMON_BRANCH_NAMES:
+            rule_builder.smart_options([remote + '/' + branch])
+
+    rule_builder.smart_options(_COMMON_BRANCH_NAMES)
+    rule_builder.smart_options(_COMMON_REMOTE_NAMES)
+
+    # TODO Implement this properly in the git grammar
+    rule_builder.option('stash at [zero]', 'stash@{0}')
```

Comment:
I don't have time to implement this properly at the moment. I don't even know if this feature will be that useful to anyone. But i have put this here anyway i can implement this property later in another pull request if anyone thinks this is worth implementing


By implementing I mean generalise this so that you could say something like `stash at <any_integer>`

Reply:
Fine by me.